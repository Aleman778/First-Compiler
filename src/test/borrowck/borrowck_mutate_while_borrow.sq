// ignore

fn main() {
    let mut a: i32 = 5;
    let b: &i32 = &a;
    a = a + 5;
    print_int(a);
    print_int(*b);
}
