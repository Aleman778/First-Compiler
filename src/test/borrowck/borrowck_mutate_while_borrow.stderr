error: cannot assign to `a` because it is borrowed
  --> $DIR/borrowck_mutate_while_borrow.sq:6:5
   |
LL |     let b: &i32 = &a;
   |                   -- borrow of `a` occurs here
LL |     a = a + 5;
   |     ^^^^^^^^^ assignment to borrowed `a` occurs here
   |     print_int(a);
   |     print_int(*b);
   |               -- borrow later used here